### ISSUE TITLE
Spades assembly crashed without log error

### ISSUE THREAD

**Issue by Fla1487** [rrwick/Unicycler#334]
Dear All,
I have a problem with unicylcer and short reads alignment by using spades.
I have previously used the same command without any troubles. Now with new sequences, and old ones, unicycler crashs.

Unicycler version: v0.5.0
Using 16 threads

Dependencies:
  Program       Version   Status  
  spades.py     3.15.5    good    
  racon                   not used
  makeblastdb   2.14.1+   good    
  tblastn       2.14.1+   good    

This is the generated error (I have checked the log file without finding anything):

Error: SPAdes encountered an error:

Could You help to identify the problem?

Thank you in advance

**Comment by Fla1487** [comment]
I add also that follwoing uniclycler/spades guides, it is possible modify spades parameter, so that I added --spades options "-m 1024".
Unfortunatly I obtained:
unicycler: error: unrecognized arguments: --spades-options -m 1024

Honestly, I do not understand whats is the problem

Thank you again

**Comment by pwkooij** [comment]
@Fla1487 Not sure if it will resolve your problems, but I believe the correct argument should be --spades_options -m 1024

**Comment by marrip** [comment]
I have the same problem, did not try the option though...

**Comment by anaalmeidasantos** [comment]
Anyone manage to fix this? I got the same thing.

**Comment by pwkooij** [comment]
> Anyone manage to fix this? I got the same thing.

Did you check whether the command was spelled correctly? (see my comment above)

**Comment by anaalmeidasantos** [comment]
Yes! I do this all the time and have only recently started encountering this error. 
![Screenshot 2024-07-08 at 15 51 28](https://github.com/rrwick/Unicycler/assets/122802203/695102b4-1d2b-4a56-b38c-8372fa37c7cf)

**Comment by pwkooij** [comment]
Can you show what the exact error is SPAdes is giving? It might be an issue with the swap space, which I had previously as well -> https://github.com/rrwick/Unicycler/issues/300#issuecomment-2142124082

**Comment by anaalmeidasantos** [comment]
The error is the following:

WARNING: Try to use logger before create one. Level=ERROR. Message=File /hpc/dla_mm/anasantos/data/202301_Efaecalis/data/hybrid_genomes/M508826_hybrid/spades_assembly/K111/configs/config.info doesn't exist or can't be read!


== Error ==  system call for: "['/hpc/dla_mm/anasantos/data/miniconda3/envs/unicycler_env/bin/spades-core', '/hpc/dla_mm/anasantos/data/202301_Efaecalis/data/hybrid_genomes/M508826_hybrid/spades_assembly/K111/configs/config.info', '/hpc/dla_mm/anasantos/data/202301_Efaecalis/data/hybrid_genomes/M508826_hybrid/spades_assembly/K111/configs/isolate_mode.info']" finished abnormally, OS return value: 2

Something to do with the config.info file?

**Comment by pwkooij** [comment]
@anaalmeidasantos This looks similar to the error I had. Try increasing your SWAP, that might resolve it (it did for me)

**Comment by anaalmeidasantos** [comment]
I'm sorry, but I'm not very experienced with this. Could you guide me on how to increase my SWAP?

**Comment by pwkooij** [comment]
Are you running it locally or on a server?

**Comment by anaalmeidasantos** [comment]
On a server.

**Comment by pwkooij** [comment]
ok, then I have no idea, and not sure if that is the problem. Might be better to post the problem on the SPAdes github, since it's related to that.

I was running locally

**Comment by anaalmeidasantos** [comment]
Thank you for your help!

**Comment by niamhlacy** [comment]
Hi! I had the same problem and using using SPAdes 3.15.4 and python 3.9 solved it :)

**Comment by victor5lm** [comment]
Hi,

just as I indicated in [this comment](https://github.com/rrwick/Unicycler/issues/351#issuecomment-5437084296), I had a similar issue (SPAdes failing but not reporting the error at all) running [nf-core/bacass](https://github.com/nf-core/bacass/tree/2.6.1) using unicycler v5.0.1 (being launched using SPAdes v4.1.0):
`Error: SPAdes encountered an error:`

This was fixed in my case running unicycler v5.0.1 with the last version of SPAdes (currently, v4.3.0), for which the developers reported to have improved overall RAM consumption for all pipelines and introduced the proper error-code system with clear exit codes and error messages, fixing the error handling mechanism.
